src/domain/products/product-form/sections/sales-channels.tsx

```diff
@@ -19,46 +23,83 @@ const SalesChannels: React.FC<SalesChannelsProps> = ({
   isEdit = false,
   product,
 }) => {
-  const { count } = useAdminSalesChannels()
-  const remainder = Math.max(product?.sales_channels?.length - 3, 0)
+  const {
+    state: isScModalOpen,
+    open: openScModal,
+    close: closeScModal,
+  } = useToggleState()
+
+  const { salesChannels, setSalesChannels } = useProductForm()
+
+  const { count, isLoading: isLoadingSalesChannels } = useAdminSalesChannels()
+  const { store, isLoading: isLoadingStore } = useAdminStore()
 
+  const isLoading = isLoadingSalesChannels || isLoadingStore
+
+  useEffect(() => {
+    if (!isEdit && !isLoadingStore) {
+      setSalesChannels([store?.default_sales_channel as SalesChannel])
+    }
+  }, [isLoadingStore])
+
+  const remainder = Math.max(product?.sales_channels?.length - 3, 0)
   return (
-    <BodyCard
-      title="Sales Channels"
-      subtitle="This product will only be available in the default sales channel if left untouched."
-      className="min-h-[200px]"
-    >
-      <div className="flex space-x-2">
-        <div className="flex space-x-2 max-w-[600px] overflow-clip">
-          {isEdit &&
-            product.sales_channels
-              ?.slice(0, 3)
-              .map((sc) => <SalesChannelBadge channel={sc} />)}
-        </div>
-        {remainder > 0 && (
-          <Badge variant="ghost">
-            <div className="flex items-center h-full inter-base-regular text-grey-50">
-              + {remainder} more
+    <>
+      <BodyCard
+        title="Sales Channels"
+        subtitle="This product will only be available in the default sales channel if left untouched."
+        className="min-h-[200px]"
+      >
+        {isLoading ? (
+          <Spinner />
+        ) : (
+          <>
+            <div className="flex space-x-2">
+              <div className="flex space-x-2 max-w-[600px] overflow-clip">
+                {salesChannels?.slice(0, 3).map((sc) => (
+                  <SalesChannelBadge channel={sc} />
```

Comment:
todo: The sales channel badge is missing the cross to remove it directly from within the section (It is in the design)

Reply:
comment: also, the default sales channel should be always position as the first element like we said in the settings. wdyt?

Reply:
removed it due to @olivermrbl's comment: https://github.com/medusajs/admin/pull/647#discussion_r924699507

Reply:
@olivermrbl I still think that feature is nice to have to avoid having to open to modal to remove one channel and therefore allow to reduce the number of actions to succeed removing one channel or just quick removal. wdyt guys?